### PR TITLE
[NEW BONUS GUIDE] JoinMarket-webui (Jam)

### LINKED ISSUE #1098: [FEATURE REQ] Add guide for Jam (web UI for JoinMarket)
### What & why

[Jam](https://github.com/joinmarket-webui/jam) is a user friendly web UI for JoinMarket. It's already available for RaspiBlitz, Umbrel and Citadel. Would be cool to have Jam guide for RaspiBolt as well.

### How

First JoinMarket guide for RaspiBolt v3 should be merged (#1085). Then either instructions for Jam should be added to that guide or separate guide should be written.

### PR DESCRIPTION
**Pull request by nu5h4ckle**
#### What

- Created JoinMarket-webui (Jam) bonus guide

### Why

- Jam enables a simple UX for interacting with JoinMarket-clientserver

#### How

- Reviewing and conforming to existing bonus guides

#### Scope

- [ ] significant change to core configuration
- [x] independent bonus guide
- [ ] simple bug fix

Fixes #1098 

#### Test & maintenance

How can the changes be tested? Is there ongoing maintenance effort? If this is a bonus guide: are you willing to update it from time to time?

- Test by following guide and reporting issues.  I will make an effort to keep the guide up-to-date.

#### Animated GIF (optional)

Add some snazz if you feel like it :)

### CODE REVIEW

**Comment by Buttafuoco** [comment]
i've just installed but do not able to connect. see this issue

https://github.com/joinmarket-webui/jam/issues/571

someone say where i can dig deeper to find the solution?

File: guide/bonus/Jam.md [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+```sh
+$ sudo systemctl restart tor
+```
+
+### Remove Nginx configuration
+
+* Remove Nginx configuration for Jam
+
+```sh
+$ sudo rm /etc/nginx/sites-enabled/dojo.conf
```
Comment (theborakompanioni) [inline guide/bonus/Jam.md:426]: Should this be
```suggestion
$ sudo rm /etc/nginx/streams-enabled/jam-reverse-proxy.conf
```
?

File: guide/bonus/Jam.md [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+```
+
+* Uncomment these two separate lines, set desired values, then save
+```sh
+max_cj_fee_rel = 0.00002
+
+max_cj_fee_abs = 500
+```
+
+* Generate self-signed certificate in JoinMarket's working directory
```
Comment (Buttafuoco) [inline guide/bonus/Jam.md:99]: cd /home/joinmarket/.joinmarket (it's not clear what folder is this)

**Review by Buttafuoco — APPROVED** [review]

**Review by greenart7c3 — CHANGES_REQUESTED** [review]

File: guide/bonus/bitcoin/Jam.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+```
+
+* Generate self-signed certificate in JoinMarket's working directory
+
+```sh
+$ mkdir ssl/ && cd "$_"
+
+$ openssl req -newkey rsa:4096 -x509 -sha256 -days 3650 -nodes \
+  -out cert.pem -keyout key.pem \
+  -subj "/C=US/ST=Utah/L=Lehi/O=Your Company, Inc./OU=IT/CN=example.com"
```
Comment (kristapsk) [inline guide/bonus/bitcoin/Jam.md:106]:
```suggestion
  -subj "/C=US/ST=Utah/L=Lehi/O=Your Company, Inc./OU=IT/CN=example.com"
$ exit
```

Need to switch back to admin user for next commands.

File: guide/bonus/bitcoin/Jam.md [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+$ sudo npm install latest-version
+```
+
+### JoinMarket preparations
+* Provide values for config variables `max_cj_fee_abs` and `max_cj_fee_rel` in `joinmarket.cfg`. Set them to values you feel comfortable with.
+
+```sh
+$ sudo su - joinmarket
+$ cd .joinmarket
+$ nano joinmarket.cfg
```
Comment (kristapsk) [inline guide/bonus/bitcoin/Jam.md:89]:
```suggestion
$ nano ~/.joinmarket/joinmarket.cfg
```

File: guide/bonus/bitcoin/Jam.md [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+```
+
+* Uncomment these two separate lines, set desired values, then save
+```sh
+max_cj_fee_rel = 0.00002
+
+max_cj_fee_abs = 500
```
Comment (kristapsk) [inline guide/bonus/bitcoin/Jam.md:96]: I would increase these values. 0.00002 and 500 are yield generator configuration defaults, but with 0.1 (10%) variance. Would suggest increasing to at least 0.00003 and 600. Lower values means you may save some satoshis, but it potentially lowers your anonymity set, as there will be less makers to choose from when doing coinjoins as a taker.

File: guide/bonus/bitcoin/Jam.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+* Obtain developer pgp signature for verification
+
+```sh
+$ curl https://dergigi.com/PGP.txt | gpg --import
+```
+
+* Retrieve source code
+
+```sh
+$ git clone https://github.com/joinmarket-webui/jam.git --branch v0.1.3 --depth=1
```
Comment (kristapsk) [inline guide/bonus/bitcoin/Jam.md:173]: There have been new release.
```suggestion
$ git clone https://github.com/joinmarket-webui/jam.git --branch v0.1.4 --depth=1
```

File: guide/bonus/bitcoin/Jam.md [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+```
+
+* Verify release by looking for `Good signature` response
+
+```sh
+$ cd jam
+
+$ git verify-tag v0.1.3
```
Comment (kristapsk) [inline guide/bonus/bitcoin/Jam.md:181]: There have been new release.
```suggestion
$ git verify-tag v0.1.4
```

File: guide/bonus/bitcoin/Jam.md [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+> Good signature...
```
Comment (kristapsk) [inline guide/bonus/bitcoin/Jam.md:182]: It's a good idea to show key fingerprint here, so user can compare, does it match.
```suggestion
...
> gpg: Good signature from "Gigi <dergigi@pm.me>" [unknown]
> gpg: WARNING: This key is not certified with a trusted signature!
> gpg:          There is no indication that the signature belongs to the owner.
> Primary key fingerprint: 8198 A185 30A5 22A0 9561  2439 89C4 A25E 69A5 DE7F
```

**Comment by kristapsk** [comment]
Link needs to be added to https://github.com/raspibolt/raspibolt/blob/master/guide/bonus/bitcoin/index.md.

**Review by kristapsk — APPROVED** [review]
ACK b088b71755ebd1553746cadbbf48a5f0486134ba. I think it's good for merging now.

**Comment by SpartanBit** [comment]
Trying to run it but without success, due to this issue: 
https://github.com/joinmarket-webui/jam/issues/608
Any advice on how to solve it?

**Pull request merged**